server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -437,27 +437,61 @@ private String sanitizeMatchesFilter(String matches) {
             List<String> entitlementIds = criteria.list();
 
             if (entitlementIds != null && !entitlementIds.isEmpty()) {
-                criteria = this.currentSession()
-                    .createCriteria(Entitlement.class)
-                    .add(CPRestrictions.in("id", entitlementIds));
+                entitlementsPage = createPage(this.findEntitlements(entitlementIds), pageRequest);
```

Comment:
We're still broken here (and, for the record, we were broken before): as soon as we re-query here, we lose any sorting from the pageRequest unless we pass the same sorting params through.

Additionally, if we're going to keep the multi-query behavior, we should use this as an opportunity to not query the entire set of entitlements, and only grab the entitlements from the block of IDs we're going to fetch.